iron-list.html

```diff
@@ -908,6 +932,27 @@
 
     _scrollHandler: function() {
       this._refresh();
+
+      this.debounce('scrollThreshold', function() {
+        var scrollTop = this._scrollPosition;
+        if (scrollTop <= this.upperThreshold) {
+          if (!(this._scrollThreshold & 1)) {
+            this.fire('upper-trigger');
+            this._scrollThreshold = this._scrollThreshold | 1;

```

Comment:
Is the use of the bitmask here only to save an extra boolean?  If so, I'd recommend just using two booleans since the cost savings really isn't worth the loss of readability.